server/sonar-server/src/main/java/org/sonar/server/branch/ws/ListAction.java

```diff
@@ -148,14 +153,52 @@ private static void addBranch(ProjectBranches.ListWsResponse.Builder response, B
   private static void setBranchStatus(ProjectBranches.Branch.Builder builder, BranchDto branch, @Nullable LiveMeasureDto qualityGateMeasure,
                                       @Nullable BranchStatistics branchStatistics) {
     ProjectBranches.Status.Builder statusBuilder = ProjectBranches.Status.newBuilder();
-    if (branch.getBranchType() == LONG && qualityGateMeasure != null) {
+    if (LONG.equals(branch.getBranchType()) && qualityGateMeasure != null) {
       Protobuf.setNullable(qualityGateMeasure.getDataAsString(), statusBuilder::setQualityGateStatus);
     }
-    if (branch.getBranchType() == BranchType.SHORT) {
+    if (SHORT.equals(branch.getBranchType())) {
       statusBuilder.setBugs(branchStatistics == null ? 0L : branchStatistics.getBugs());
       statusBuilder.setVulnerabilities(branchStatistics == null ? 0L : branchStatistics.getVulnerabilities());
       statusBuilder.setCodeSmells(branchStatistics == null ? 0L : branchStatistics.getCodeSmells());
     }
     builder.setStatus(statusBuilder);
   }
+
+  private static void addPullRequest(ProjectBranches.ListWsResponse.Builder response, BranchDto branch, Map<String, BranchDto> mergeBranchesByUuid,
+                                     BranchStatistics branchStatistics, @Nullable String analysisDate) {
+    ProjectBranches.PullRequest.Builder builder = toPullRequestBuilder(branch, Optional.ofNullable(mergeBranchesByUuid.get(branch.getMergeBranchUuid())));
+    setPullRequestStatus(builder, branchStatistics);
+    if (analysisDate != null) {
+      builder.setAnalysisDate(analysisDate);
+    }
+
+    response.addPullRequests(builder);
+  }
+
+  private static ProjectBranches.PullRequest.Builder toPullRequestBuilder(BranchDto branch, Optional<BranchDto> mergeBranch) {
+    ProjectBranches.PullRequest.Builder builder = ProjectBranches.PullRequest.newBuilder();
+    String branchKey = branch.getKey();
+    setNullable(branchKey, builder::setBranch);
+    //TODO put real data
+    setNullable(branchKey, builder::setId);
+    //TODO put real data
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fteryk%2Flist-pr&issues=AWF8sv0SHTiOwYNMzcIr&open=AWF8sv0SHTiOwYNMzcIr)

[ISSUE_KEY]: AWF8sv0SHTiOwYNMzcIr